homeassistant/components/honeywell/__init__.py

```diff
@@ -43,15 +43,30 @@ async def async_setup_entry(hass, config):
         _LOGGER.debug("No devices found")
         return False
 
-    data = HoneywellData(hass, client, username, password, devices)
+    data = HoneywellData(hass, config, client, username, password, devices)
     await data.async_update()
     hass.data.setdefault(DOMAIN, {})
     hass.data[DOMAIN][config.entry_id] = data
     hass.config_entries.async_setup_platforms(config, PLATFORMS)
 
+    config.async_on_unload(config.add_update_listener(update_listener))
+
     return True
 
 
+async def update_listener(hass, config) -> None:
+    """Update listener."""
+    await hass.config_entries.async_reload(config.entry_id)
+
```

Comment:
```suggestion
```

We shouldn't reload the integration unless the configuration has changed.
It looks like you are already recreating the client in `_retry`. If something else needs to be
done to reconnect, it should be done there.

Reply:
If you need to recreate the devices on retry, move the block on line 35 into its own function and run it in the retry function

Reply:
The challenge is when we hit the retry method and recreate the client, we then recreate and reassign the devices array in `HoneywellData`. This newly reassigned array of devices (`self.devices`) does not seem to get picked up by HA. If TCC client connection fails for whatever reason and we need to retry, the climate devices all freeze in time and never see any new temp readings and you cannot set temps or modes on them. 

That being said, this was just the first way I was able to find to force HA to reload the entities for honeywell. Perhaps there is a proper way to do this? Or perhaps the way we're reading `self.devices` is incorrect?

Reply:
Actually reloading the integration is not a bad idea. All entities will be written as unavailable and the initial connect logic will kick in with automatic retries